pkg/agent/clean/adunmigration/rtbs.go

```
@@ -392,6 +467,27 @@ func migrateGRBs(workunit *migrateUserWorkUnit, sc *config.ScaledContext, dryRun
 			})
 			if err != nil {
 				logrus.Errorf("[%v] permanent error when deleting GRB, giving up: %v", migrateGrbsOperation, err)
+				continue
+			}
+
+			// Adding the annotation here has the side effect of triggering a downstream RBAC sync for our newly created
+			// binding, which is why we perform it last
+			newAnnotations[adGUIDMigrationAnnotation] = oldGrb.UserName
+			err = wait.ExponentialBackoff(backoff, func() (finished bool, err error) {
+				updatedGrb, err := sc.Management.GlobalRoleBindings("").GetNamespaced(oldGrb.Namespace, createdGrb.Name, metav1.GetOptions{})
```

Looks like err is not being checked here